unit_test/entities/test_pipe.py

```diff
@@ -64,7 +109,18 @@ def test_pipe_get_property_diameter(self):
     def test_pipe_get_property_diameter_failed(self):
         """Evaluate failure to retrieve diameter from ESDL asset."""
         # Arrange
-        pipe = Pipe(asset_name="pipe", asset_id="pipe_id", port_ids=["test1", "test2"])
+        pipe = Pipe(
+            asset_name="pipe",
+            asset_id="pipe_id",
+            port_ids=["test1", "test2"],
+            length=self.length,
+            inner_diameter=self.inner_diameter,
+            roughness=self.roughness,
+            alpha_value=self.alpha_value,
+            minor_loss_coefficient=self.minor_loss_coefficient,
+            external_temperature=self.external_temperature,
+            qheat_external=self.qheat_external,
+        )
         esdl_asset_mock = Mock()
         esdl_asset_mock.get_property.return_value = (1.0, False)
```

Comment:
This test should be moved to the test of the mapper of the pipe.

Reply:
I can't find the test of the pipe mapper.

Reply:
You need to create it yourself.

Reply:
I have added it.